### PR TITLE
feature: support sealer rmi

### LINKED ISSUE #8: add  sealer rmi command
images remove feature, like `docker rmi ` command  "Remove one or more images"

### PR DESCRIPTION
**Pull request by touchuyht**
### Describe what this PR does / why we need it
Support deleting local images. Some image may be too old or is not needed anymore, in there cases the images has to be deleted.

### Does this pull request fix one issue?

Fixes #8
### Describe how you did it
I modified some files in image pkg and add corresponding commands to cobra. Add a delete method in layerStore interface, a delete method in Metaservice interface and Service interface and implements them.

### CODE REVIEW

File: image/store/layer_store.go [hunk 1]
```diff
@@ -54,6 +54,24 @@ func (ls *layerStore) RegisterLayerIfNotPresent(closer io.ReadCloser, id LayerID
 	return nil
 }
 
+func (ls *layerStore) Delete(id LayerID) error {
+	layer := ls.Get(id)
+	if layer == nil {
+		logger.Debug("layer %s is already deleted", id)
+		return nil
+	}
+
```
Comment (justadogistaken) [inline image/store/layer_store.go:63]: digs := digest.Digest(id) here, which could be used twice

File: image/store/layer_store.go [hunk 2]
```diff
@@ -54,6 +54,24 @@ func (ls *layerStore) RegisterLayerIfNotPresent(closer io.ReadCloser, id LayerID
...
+	filePath := filepath.Join(common.DefaultLayerDir, digest.Digest(id).Hex())
+	utils.CleanDir(filePath)
```
Comment (justadogistaken) [inline image/store/layer_store.go:65]: I guess we may still need catch the err, if cleanDir failed, the layer stilll exitss

File: image/base_image_manager.go [hunk 3]
```diff
@@ -42,6 +42,20 @@ func (bim BaseImageManager) syncImageLocal(image v1.Image) (err error) {
 	return nil
 }
```
Comment (justadogistaken) [inline image/base_image_manager.go:44]: BaseImageManager aims to provide common abilities. I support implement deleteImageLocal within defaultImageService is good. Or implement a imageStore like layerStore(not urgent)

File: image/default_image.go [hunk 4]
```diff
@@ -122,6 +123,76 @@ func (d DefaultImageService) Login(RegistryURL, RegistryUsername, RegistryPasswd
...
+		if err != nil {
+			continue
+		}
+		images = append(images, tmpImage)
+	}
+	layer2ImageNames := layer2ImageMap(images)
+	layerStore, err := store.NewDefaultLayerStore()
+	if err != nil {
+		return err
+	}
```
Comment (justadogistaken) [inline image/default_image.go:159]: maybe TODO here, find a atomic way to delete layers and image.

File: image/default_image.go [hunk 5]
```diff
@@ -122,6 +123,76 @@ func (d DefaultImageService) Login(RegistryURL, RegistryUsername, RegistryPasswd
...
+		layerID := store.LayerID(digest.NewDigestFromEncoded(digest.SHA256, layer.Hash))
+		if isLayerDeletable(layer2ImageNames, layerID) {
+			err = layerStore.Delete(layerID)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	err = d.deleteImageLocal(image.Name, image.Spec.ID)
```
Comment (justadogistaken) [inline image/default_image.go:170]: deleting image before deleting could be better? in case that some of layers deletion failed

**Review by justadogistaken — CHANGES_REQUESTED** [review]

File: image/base_image_manager.go (hunk 3 above)
Comment (touchuyht) [reply image/base_image_manager.go:45]: The baseimageManager's implementation is just like a imageStore. So I consider providing deleteImageLocal in BaseImageManager is reasonable.

Comment (justadogistaken) [reply image/base_image_manager.go:45]: agree

**Review by justadogistaken — APPROVED** [review]
LGTM

**Review by justadogistaken — APPROVED** [review]

**Pull request merged**